crates/sui-indexer/src/handlers/committer.rs

```diff
@@ -212,6 +219,17 @@ async fn commit_checkpoints<S>(
         })
         .expect("Persisting data into DB should not fail.");
 
+    state
```

Comment:
- Should this happen after the code guarded by `if is_epoch_end` below? It seems like that should be included within the watermark.
- Is it an issue that persisting checkpoints and updating the watermark are not done in a single transaction? I.e. what happens if the indexer crashes just after persisting the checkpoint. For example, if the write to the watermark itself fails?